### PR TITLE
qt5: fix splicing

### PR DESCRIPTION
**Pull request by Artturin**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by amjoseph-nixpkgs** [comment]
Working on testing this.

Here is the qt6 version: https://github.com/NixOS/nixpkgs/pull/264959

**Comment by amjoseph-nixpkgs** [comment]
I'll try to rebase https://github.com/NixOS/nixpkgs/pull/264965 on this

**Comment by Artturin** [comment]
This reduces the amount of derivations to build after building all the succeeding ones

`nix build ".#pkgsCross.aarch64-multiplatform.qt5.qt3d" --keep-going`

```
these 14 derivations will be built:
  /nix/store/jk3dv2r7c750949k8l5k40l7qgavqbyq-qttranslations-5.15.10.drv
  /nix/store/789zvlpcdl5gjdzzmw0j4izin9v6jp9c-qtbase-5.15.10.drv
  /nix/store/niag4vfnhpmdlms6sfqs9mj0444c5q0g-qmake-hook.drv
  /nix/store/v9svbj6mhmq5xr7hims7jwd84ilhh2cn-qtsvg-5.15.10.drv
  /nix/store/k9nd4ls8ajsvza6y1rh8gxh69k3xgkgl-qtdeclarative-5.15.10.drv
  /nix/store/qdl8pwnygkwygrimbwzr71x5b56lqwcd-qttools-5.15.10.drv
  /nix/store/79ci43i5jpwd4igk3wzxjlr6awn6n4dg-qtbase-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/r9dk0rq6wr48ijv2vq6r1vg5mb9varxv-qmake-hook.drv
  /nix/store/8v9z215v19j1x0fal0116xjpd8addhlh-qttranslations-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/9p4dgv9179mxfb4gd713v58g93c0dcc5-qtbase-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/fb8dbral8x8hdqrdnylmybxvkcsyzkip-qmake-hook.drv
  /nix/store/60g2ws7qgcfjj65bcvgn8xlrnlsrs9xq-qtsvg-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/lhgxqvx5xsgdzh8apbdiq8zdj4lvpzah-qtdeclarative-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/fag3a6ldrqdr47vhra583c5cyms5ax60-qt3d-aarch64-unknown-linux-gnu-5.15.10.drv
```


```
these 8 derivations will be built:
  /nix/store/m5lsbhw0796r9p35zfva78sbg7z3am31-qtsvg-5.15.10.drv
  /nix/store/41jg69xbvnvdyn1ayh80alqzy8nznb4l-qtdeclarative-5.15.10.drv
  /nix/store/wdg936p947wd4gx87pn1arqxk74p38sw-qttools-5.15.10.drv
  /nix/store/2k3dwsza8rv8ilasaxbfr1ar2mmgfrs2-qttranslations-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/pwzf1pwq1h43bq720dmxfng8g663h29v-qtbase-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/df1h97nqhdkfw4ns75hq4grjkqwg5n1a-qtsvg-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/kbmxzwbc9xm8c2x4q7vaw2yzv9z4jd6k-qtdeclarative-aarch64-unknown-linux-gnu-5.15.10.drv
  /nix/store/c2sxcimrl9nxaw4jncxll2sh0fg3lb3c-qt3d-aarch64-unknown-linux-gnu-5.15.10.drv
```

14 vs 8

**Review by ghost — APPROVED** [review]
We should merge this soon.

**Comment by Artturin** [comment]
`pkgsCross.aarch64-multiplatform.qt5.full`
41 failed to 35 failed

**Comment by amjoseph-nixpkgs** [comment]
I have the fix.  There was a rebase issue with the path to meson's `emulator-hook.sh` which got moved into `pkgs/by-name`.

**Comment by amjoseph-nixpkgs** [comment]
https://github.com/NixOS/nixpkgs/pull/265134

File: pkgs/development/libraries/qt-5/5.15/default.nix [hunk 1]
```diff
@@ -318,20 +311,20 @@ let
...
-      } ../hooks/qmake-hook.sh;
+      } ../hooks/qmake-hook.sh) { };
 
-      wrapQtAppsHook = makeSetupHook {
+      wrapQtAppsHook = callPackage ({ makeBinaryWrapper, qtbase, qtwayland }: makeSetupHook {
         name = "wrap-qt5-apps-hook";
-        propagatedBuildInputs = [ self.qtbase.dev buildPackages.makeBinaryWrapper ]
-          ++ lib.optional stdenv.isLinux self.qtwayland.dev;
-      } ../hooks/wrap-qt-apps-hook.sh;
+        propagatedBuildInputs = [ qtbase.dev makeBinaryWrapper ]
```
Comment (amjoseph-nixpkgs) [inline pkgs/development/libraries/qt-5/5.15/default.nix:325]: Something still isn't right here.

- https://github.com/NixOS/nixpkgs/pull/266375

**Pull request merged**